packages/salesforcedx-vscode-apex/src/testDiscovery/testDiscoveryService.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import type { ToolingTestClass, TestDiscoveryResult, ToolingTestsPage } from './schemas';
+import type { Connection } from '@salesforce/core';
+import { getVscodeCoreExtension } from '../coreExtensionUtils';
+import { getTelemetryService } from '../telemetry/telemetry';
+
+/**
+ * Service that retrieves Apex test classes and methods from the Tooling REST Test Discovery API.
+ * Docs: https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/intro_rest_resources_testing_discovery.htm
+ */
+export class TestDiscoveryService {
+  private connection: Connection;
+
+  constructor(connection: Connection) {
+    this.connection = connection;
+  }
+
+  public static async create(): Promise<TestDiscoveryService | undefined> {
+    try {
+      const core = await getVscodeCoreExtension();
+      const connection = await core.exports.services.WorkspaceContext.getInstance().getConnection();
```

Comment:
you'll want to force the connection to a high enough API since this service is only available >=65

Reply:
@mshanemc I have the same need in the OAS PR, getting max api version from the org. There is a prop on authInfo (instanceApiVersion) that seems to fit the bill. Do you recall if that property is guaranteed to be org version?